From 7f7fda4f3f4d4af8b30ce73337ef0850d59d7429 Mon Sep 17 00:00:00 2001
From: Egor Pasko <pasko@google.com>
Date: Thu, 1 Jul 2010 16:15:12 +0200
Subject: [PATCH 22/22] [gcc] Allow GCC insert .cfi_startproc and .cfi_endproc.

NaCl GAS supports these assembler directives. Though on Mac buildbots this
capability is not recognized due to the lack of readelf installed. Supporting
these directives is essential to have correct backtraces listed with valgrind.

BUG: http://code.google.com/p/nativeclient/issues/detail?id=663

Review URL: http://codereview.chromium.org/2805058
---
 gcc/gcc/config/i386/nacl.h |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/gcc/gcc/config/i386/nacl.h b/gcc/gcc/config/i386/nacl.h
index 2a8bb97..4b3d734 100644
--- a/gcc/gcc/config/i386/nacl.h
+++ b/gcc/gcc/config/i386/nacl.h
@@ -240,6 +240,11 @@ Boston, MA 02111-1307, USA.  */
 #undef HAVE_AS_GOTOFF_IN_DATA
 #define HAVE_AS_GOTOFF_IN_DATA 1
 
+/* Configure script incorrectly detects HAVE_GAS_CFI_DIRECTIVE when readelf is
+ * not found in PATH. */
+#undef HAVE_GAS_CFI_DIRECTIVE
+#define HAVE_GAS_CFI_DIRECTIVE 1
+
 /* NaCl reserves R15 and makes RBP special in x86-64 mode */
 #undef FIXED_REGISTERS
 #define FIXED_REGISTERS						\
-- 
1.7.1

